### ISSUE TITLE
Tracking Issue: Remove deprecated parts of App init API without breaking examples.

### ISSUE THREAD

**Issue by Ben-PH** [seed-rs/seed#448]
# Motivation

@MartinKavik : 

> The latest API to start Seed app is `App::start("app", init, uodate, view)`. However there are old deprecated APIs still in Seed code-base - e.g. the whole app [Builder](https://github.com/seed-rs/seed/tree/master/src/app/builder).  These old starting APIs should be deleted - it sounds easy, but the code is too clever (it's also the reason to delete it). The goal - old APIs and their files are deleted and all examples still works :slightly_smiling_face:  It's blocker for some other Seed tasks and it'll improve Seed code a lot.
>The related task -  we replaced the concept of Global Msgs with subscribe/notify so all functions and item that uses Global Msgs are basically deprecated. However they are still in Seed code and complicate API and the code. The goal - remove everything related to Global Msgs - you can find them by searching for GMs or GMsg in Seed repo (e.g. [Orders trait](https://github.com/seed-rs/seed/blob/master/src/app/orders.rs#L15)). And all examples should still work after the refactor. It would be nice to add this change to Seed 0.8.0. This change will improve Seed code and also make writing Elm components easier.

# General

- [ ] add script to dev branch to ensure all examples are unbroken
- [ ] Settle on sem-var procedure
- [ ] Checklist of deprecated code to be removed.

# Ease of maintenance

- Setup scans of public crates that depend on seed, and consult with their maintainers if affected.
- Ensure all removal tasks have a dedicated github issue
- Update deprecation notes where not flagged for removal and appropriate.
- Ensure each issue provides seed users information for best ease-of-maintenance.

# Specific deprecations - to be expanded as needed

- [ ] Remove Builder API
- [ ] Remove Global Messages API's
- TODO: Add more as needed

# Specific remove checklist - to be put in each corresponding issue

- [ ] Examples do not emit errors or deprication warnings
- [ ] Scan public crates for impacted projects, with issues raised with relevant projects.

**Comment by MartinKavik** [comment]
>  add script to dev branch to ensure all examples are unbroken

`cargo make verify` compiles and lints all examples. However manual testing on multiple platforms and browsers is preferred if you want to be sure that it really works.

> Settle on sem-var procedure

We use "pre-stable sem-ver"` => `0.x.y`, where `x` is incremented where there are breaking changes, otherwise only `y` is incremented. One stable, we'll move to standard semver. (There is issue about it somewhere).

> Deprecate Builder API
> Deprecate Global Messages API's

The most of them are already deprecated (some of them for multiple Seed versions). However it became too messy to continue and the complexity blocks further development. It's time for breaking change.

**Comment by Ben-PH** [comment]
understood. Ill edit and remove as needed.

**Comment by AlterionX** [comment]
The original author of the wacky Builder API here. Cheers on removing it! It kind of spiraled out of control while attempting to maintain backwards compat so that deprecation could work. Would it be possible to ping me with the reviews, so that I can keep up with what's going on?

**Comment by Ben-PH** [comment]
Sure thing! 

I got a bit too fancy, I think, with this issue. first non-trivial contribution so over-compensated. A draft PR of my work in progress is in PR #462 

If you're happy to, it would be of some help to talk through the draft commit-by-commit. I'm thinking that we can look at how I went about unwinding the changes, and see if I've missed or done something that could come back to bite us later. The number of commits makes it look a lot more work than it actually is - each one is quite small, and many of them are things like formatting and comment changes. 

Good to see you drop in.
